pySNOM/images.py

```diff
@@ -1,118 +1,121 @@
-        else:
-            m = self.parameters["Scan"]
-            self.setMeasurementMode(defaults.image_mode_defs[m])
-
-    def getImageFromChannel(self, channelname: str):
-        singleimage = Image()
 
-        singleimage.filename = self.filename
-        singleimage.mode = self.mode
-        singleimage.parameters = self.parameters
-        singleimage.channel = channelname
-
-        channeldata = self.getChannelData(channelname)
-        singleimage.setImageParameters(channeldata)
-        singleimage.setChannel(channelname=channelname)
+    def image_from_channel(self, channelname: str):
+        """Returns a single Image object with the requred channeldata"""
+        channeldata = self.extract_channel(channelname)
+        singleimage = Image(filename = self.filename, data = channeldata, mode = self.mode, channel = channelname, info = self.info)
 
         return singleimage
 
 
 # Single image from a single data channel
-class Image(Data):
-    def __init__(self, channeldata = None) -> None:
+class Image(Measurement):
+    def __init__(self, filename = None, data = None, mode = "AFM", channelname = 'Z raw', order = int(0), datatype = DataTypes['Topography']):
         super().__init__()
         # Describing channel and datatype
-        self.channel = None # Full channel name
-        self.order = None   # Order, nth
-        self.datatype = None # Amplitude, Phase, Topography - Enum DataTypes
-        # Important measurement parameters from gwyddion fields
-        self.xreal = None   # Physical image width
-        self.yreal = None   # Physical image height
-        self.xoff = None    # Center position X
-        self.yoff = None    # Center position Y
-        self.xres = None    # Pixel size in X
-        self.yres = None    # Pixel size in Y
-        # Overwrite the data definition of parent class
-        self.data = None # Actual data, this case it is only a single channel
-        self.processor = Process()
-
-        if channeldata:
-            self.setImageParameters(channeldata)
-
-    # Method to set the actual data (numpy array) - overwrite parent class method
-    def setData(self, data):
+        self.channel = channelname # Full channel name
+        self.order = int(order)   # Order, nth
+        self.datatype = datatype # Amplitude, Phase, Topography - Enum DataTypes
+        self.data = data # Actual data, this case it is only a single channel
+
+    @property
+    def data(self):
+        """Property - data (numpy array)"""
         # Set the data
-        self.data = data
-        self.processor.setData(self.data)
-        
-    def getData(self):
-        # Set the data
-        return self.data
+        return self._data
     
-    def setChannel(self, channelname):
-        self.channel = channelname
-        self.order = int(self.channel[1])
+    @data.setter
+    def data(self, value):
+        # Set the data and the corresponding image attributes
+        if value is GwyDataField:
```

Comment:
If I see a condition this specific, this probably means that you should have two subclasses.

Reply:
I am not sure how do you mean. Like gwyimage and ndimage subclass or how?

Reply:
Yes, if one data set is special it can be a subclass of this one.